src/index/diskann/diskann.cc

```diff
@@ -520,6 +551,58 @@ DiskANNIndexNode<DataType>::Deserialize(const BinarySet& binset, std::shared_ptr
     return Status::success;
 }
 
+template <typename DataType>
+expected<std::vector<IndexNode::IteratorPtr>>
+DiskANNIndexNode<DataType>::AnnIterator(const DataSetPtr dataset, std::unique_ptr<Config> cfg,
+                                        const BitsetView& bitset) const {
+    if (!is_prepared_.load() || !pq_flash_index_) {
+        LOG_KNOWHERE_ERROR_ << "Failed to load diskann.";
+        return expected<std::vector<std::shared_ptr<IndexNode::iterator>>>::Err(Status::empty_index,
+                                                                                "DiskANN not loaded");
+    }
+
+    auto search_conf = static_cast<const DiskANNConfig&>(*cfg);
+    if (!CheckMetric(search_conf.metric_type.value())) {
+        return expected<std::vector<std::shared_ptr<IndexNode::iterator>>>::Err(Status::invalid_metric_type,
+                                                                                "unsupported metric type");
+    }
+
+    constexpr uint64_t k_lsearch_iterator = 32;
+    auto lsearch = static_cast<uint64_t>(search_conf.search_list_size.value_or(k_lsearch_iterator));
+    auto beamwidth = static_cast<uint64_t>(search_conf.beamwidth.value());
+    auto filter_ratio = static_cast<float>(search_conf.filter_threshold.value());
+
+    bool use_reorder_data = false;
+    bool for_tuning = false;
```

Comment:
these two default to `false`? Possibly pass them in from config?

Reply:
both removed.